Fix asm and global_asm macros

simonschoening opened the pull request:
The macros are now namespaced:
https://github.com/rust-lang/rust/issues/84019

wangrunji0408 commented:
Hi. Thanks for your contribution!
I just enabled CI and it failed on some other targets.
Could you take a look and try to fix them?

simonschoening committed:
Fix asm and global_asm macros

The macros are now namespaced:
https://github.com/rust-lang/rust/issues/84019

In addition, llvm_asm was removed.

simonschoening commented:
> Hi. Thanks for your contribution! I just enabled CI and it failed on some other targets. Could you take a look and try to fix them?

Thanks for your answer!
I have tried to fix some of the issues.
In addition to the namespacing, the llvm_asm macro was also removed:
https://github.com/rust-lang/rust/issues/70173

I replaced it with the asm macro, but I'm not sure whether the  implementation is correct.

simonschoening commented on src/arch/mipsel/trap.rs:
```diff
@@ -11,7 +13,10 @@ global_asm!(include_str!("trap.S"));
 pub unsafe fn init() {
     // Set cp0 ebase(15, 1) register to trap entry
-    llvm_asm!("mtc0 $0, $$15, 1":: "r"(trap_entry as usize) :: "volatile");
+    asm!(
```
Is this correct?

simonschoening commented on src/arch/mipsel/trap.rs:
```diff
@@ -11,7 +13,10 @@ global_asm!(include_str!("trap.S"));
 pub unsafe fn init() {
     // Set cp0 ebase(15, 1) register to trap entry
-    llvm_asm!("mtc0 $0, $$15, 1":: "r"(trap_entry as usize) :: "volatile");
+    asm!(
+        "mtc0 {trap_entry}, $15, 1",
+        trap_entry = in(reg) trap_entry,
+    );
```
Is this correct?

wangrunji0408 replied:
It seems right to me, but I'm not 100% sure. The MIPS target is least used, so maybe we can revisit it in the future.

wangrunji0408 reviewed (approved):
LGTM. Thanks again!
